pallets/phala/src/compute/vault.rs

```diff
@@ -392,6 +402,12 @@ pub mod pallet {
 			let mut pool_info = ensure_vault::<T>(pid)?;
 			let a = amount; // Alias to reduce confusion in the code below
 
+			if let Some(whitelist) = basepool::PoolContributionWhitelists::<T>::get(&pid) {
```

Comment:
Maybe make a function inside `basepool` to check whitelist?

Reply:
done